File: src/main/java/me/devksh930/hr/domain/model/EmployeeJobHistoryQuery.java
package me.devksh930.hr.domain.model;

import java.time.LocalDate;

public record EmployeeJobHistoryQuery(
	Integer employeeId,
	LocalDate startDate,
	LocalDate endDate,
	String jobId,
	String jobTitle,
	Integer departmentId,
	String departmentName
) {
}

File: src/main/java/me/devksh930/hr/domain/model/EmployeeJobHistoryQuery.kt
package me.devksh930.hr.domain.model

import java.time.LocalDate

data class EmployeeJobHistoryQuery(
   val employeeId: Int,
   val startDate: LocalDate,
   val endDate: LocalDate,
   val jobId: String,
   val jobTitle: String,
   val departmentId: Int,
   val departmentName: String
)
